### PR TITLE
Papa charlie/master

### PR DESCRIPTION
**Pull request by dragonsinth**

### CODE REVIEW

File: grpcurl.go [hunk 1]
```diff
@@ -697,3 +707,8 @@ func (c *errSignalingCreds) ClientHandshake(ctx context.Context, addr string, ra
 	}
 	return conn, auth, err
 }
+
+func (c *errSignalingCreds) UsesXDS() bool {
+	xc, ok := c.TransportCredentials.(interface{ UsesXDS() bool })
+	return ok && xc.UsesXDS()
+}
```
Comment (PapaCharlie) [inline grpcurl.go:714]: I don't think this is necessary. I checked through it with a debugger, `errSignalingCreds` will implement `UsesXDS` as long as the `TransportCredentials` does

**Review by PapaCharlie — CHANGES_REQUESTED** [review]

**Comment by PapaCharlie** [comment]
Oh also, it seems the `UsesXDS` interface disappeared in HEAD of go-grpc anyway, so this is totally moot!

**Comment by dragonsinth** [comment]
> Oh also, it seems the `UsesXDS` interface disappeared in HEAD of go-grpc anyway, so this is totally moot!

So we're on 1.59.0.  I scanned through the gRPC lib and it _looks_ like all of the existing uses even in that version are server-side only.  So I think maybe we don't need this on the client at all.  Can you try it out without:?

File: grpcurl.go (hunk 1 above)
Comment (PapaCharlie) [reply grpcurl.go:714]: Nevermind, you are totally right, you do need to explicitly implement the interface. The part I don't like about this in general though is that that is completely hidden in the code. It feels strange that that's how it's wired in. In the later versions of grpc-go, are there new interfaces like this?

Comment (dragonsinth) [reply grpcurl.go:714]: No clue.  See my other comment tho-- it looks to me like grpclib only cares about `UsesXDS` on the server side transport.  We're only using client, so I _think_ we could just rip this out.  But you'd need to test it for me.

**Pull request closed without merging**